File: query_kilo_db.js
const fs = require('fs');
const initSqlJs = require('sql.js');

const dbPath = 'C:\\Users\\pnv_llg.ASP\\.local\\share\\kilo\\kilo.db';

async function main() {
    const SQL = await initSqlJs();
    const fileBuffer = fs.readFileSync(dbPath);
    const db = new SQL.Database(fileBuffer);

    const sessions = db.exec(`
        SELECT s.*, p.worktree as project_worktree, p.name as project_name
        FROM session s
        LEFT JOIN project p ON s.project_id = p.id
        ORDER BY s.time_created DESC
    `);

    console.log("=".repeat(120));
    console.log(`TOTAL SESSIONS: ${sessions[0] ? sessions[0].values.length : 0}`);
    console.log("=".repeat(120));

    if (sessions[0]) {
        const cols = sessions[0].columns;
        sessions[0].values.forEach((row, i) => {
            console.log(`\n--- Session #${i+1} ---`);
            cols.forEach((col, j) => {
                const val = row[j];
                console.log(`  ${col}: ${val === null ? 'NULL' : val}`);
            });
        });
    }

    console.log("\n\n" + "=".repeat(120));
    console.log("ALL PROJECTS");
    console.log("=".repeat(120));
    const projects = db.exec("SELECT * FROM project ORDER BY time_updated DESC");
    if (projects[0]) {
        projects[0].values.forEach(row => {
            const obj = {};
            projects[0].columns.forEach((col, i) => { obj[col] = row[i]; });
            console.log(`\n  id: ${obj.id}`);
            console.log(`  name: ${obj.name}`);
            console.log(`  worktree: ${obj.worktree}`);
            console.log(`  vcs: ${obj.vcs}`);
            console.log(`  time_created: ${new Date(obj.time_created).toISOString()}`);
            console.log(`  time_updated: ${new Date(obj.time_updated).toISOString()}`);
            console.log(`  sandboxes: ${obj.sandboxes}`);
        });
    }

    console.log("\n\n" + "=".repeat(120));
    console.log("ALL PROJECT DIRECTORIES");
    console.log("=".repeat(120));
    const dirs = db.exec("SELECT * FROM project_directory ORDER BY project_id, type");
    if (dirs[0]) {
        dirs[0].values.forEach(row => {
            const obj = {};
            dirs[0].columns.forEach((col, i) => { obj[col] = row[i]; });
            console.log(`  project_id: ${obj.project_id}, directory: ${obj.directory}, type: ${obj.type}, strategy: ${obj.strategy}`);
        });
    }

    db.close();
    console.log("\nDone.");
}

main().catch(err => {
    console.error("Error:", err);
    process.exit(1);
});

File: _query_kilo_db.py
import sqlite3
import json
from datetime import datetime, timezone, timedelta

db_path = r"C:\Users\pnv_llg.ASP\.local\share\kilo\kilo.db"
conn = sqlite3.connect(db_path)
conn.row_factory = sqlite3.Row
cursor = conn.cursor()

cursor.execute("""
    SELECT s.*, p.worktree as project_worktree, p.name as project_name
    FROM session s
    LEFT JOIN project p ON s.project_id = p.id
    ORDER BY s.time_created DESC
""")
sessions = cursor.fetchall()

print("=" * 120)
print(f"TOTAL SESSIONS: {len(sessions)}")
print("=" * 120)

for i, s in enumerate(sessions):
    created = datetime.fromtimestamp(s['time_created']/1000, tz=timezone.utc).astimezone(timezone(timedelta(hours=2)))
    updated = datetime.fromtimestamp(s['time_updated']/1000, tz=timezone.utc).astimezone(timezone(timedelta(hours=2)))
    archived = None
    if s['time_archived']:
        archived = datetime.fromtimestamp(s['time_archived']/1000, tz=timezone.utc).astimezone(timezone(timedelta(hours=2)))
    
    print(f"\n--- Session #{i+1} ---")
    print(f"  id: {s['id']}")
    print(f"  slug: {s['slug']}")
    print(f"  title: {s['title']}")
    print(f"  directory: {s['directory']}")
    print(f"  path: {s['path']}")
    print(f"  project_id: {s['project_id']}")
    print(f"  project_name: {s['project_name']}")
    print(f"  project_worktree: {s['project_worktree']}")
    print(f"  workspace_id: {s['workspace_id']}")
    print(f"  parent_id: {s['parent_id']}")
    print(f"  version: {s['version']}")
    print(f"  agent: {s['agent']}")
    print(f"  model: {s['model']}")
    print(f"  time_created: {created.strftime('%Y-%m-%d %H:%M:%S')}")
    print(f"  time_updated: {updated.strftime('%Y-%m-%d %H:%M:%S')}")
    if archived:
        print(f"  time_archived: {archived.strftime('%Y-%m-%d %H:%M:%S')}")
    else:
        print(f"  time_archived: NULL")
    print(f"  cost: {s['cost']}")
    print(f"  tokens: in={s['tokens_input']}, out={s['tokens_output']}, reasoning={s['tokens_reasoning']}")
    print(f"  summary_files: {s['summary_files']}, additions: {s['summary_additions']}, deletions: {s['summary_deletions']}")

print("\n\n" + "=" * 120)
print("ALL PROJECTS")
print("=" * 120)
cursor.execute("SELECT * FROM project ORDER BY time_updated DESC")
projects = cursor.fetchall()
for p in projects:
    print(f"\n  id: {p['id']}")
    print(f"  name: {p['name']}")
    print(f"  worktree: {p['worktree']}")
    print(f"  vcs: {p['vcs']}")
    created = datetime.fromtimestamp(p['time_created']/1000, tz=timezone.utc).astimezone(timezone(timedelta(hours=2)))
    updated = datetime.fromtimestamp(p['time_updated']/1000, tz=timezone.utc).astimezone(timezone(timedelta(hours=2)))
    print(f"  time_created: {created.strftime('%Y-%m-%d %H:%M:%S')}")
    print(f"  time_updated: {updated.strftime('%Y-%m-%d %H:%M:%S')}")
    print(f"  sandboxes: {p['sandboxes']}")

print("\n\n" + "=" * 120)
print("ALL PROJECT DIRECTORIES")
print("=" * 120)
cursor.execute("SELECT * FROM project_directory ORDER BY project_id, type")
dirs = cursor.fetchall()
for d in dirs:
    print(f"  project_id: {d['project_id']}, directory: {d['directory']}, type: {d['type']}, strategy: {d['strategy']}")

conn.close()